sw/host/tests/chip/rv_dm/src/access_after_hw_reset.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+use std::time::Duration;
+
+use anyhow::{Context, Result};
+use clap::Parser;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::JtagTap;
+use opentitanlib::test_utils;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::mem::MemWriteReq;
+use opentitanlib::uart::console::UartConsole;
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "10s")]
+    timeout: Duration,
+
+    /// Path to the ELF file being tested on the device.
+    #[arg(long)]
+    firmware_elf: PathBuf,
+}
+
+fn test_access_after_hw_reset(
+    opts: &Opts,
+    transport: &TransportWrapper,
+    reset_addr: u32,
+) -> Result<()> {
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    // Enable console and wait for the message.
+    let uart = &*transport.uart("console")?;
+    uart.set_flow_control(true)?;
+    let _ = UartConsole::wait_for(uart, r"Running [^\r\n]*", opts.timeout)?;
+    let _ = UartConsole::wait_for(uart, r"Waiting for commands", opts.timeout)?;
+
+    // Check debugger is available.
+    let mut jtag = opts
+        .init
+        .jtag_params
+        .create(transport)?
+        .connect(JtagTap::RiscvTap)?;
+    assert_eq!(
+        jtag.as_raw()?.execute("$_TARGETNAME.0 curstate")?,
+        "running"
+    );
+    jtag.disconnect()?;
+
```

Comment:
The testplan for `chip_sw_rv_dm_access_after_hw_reset` says `An activation would be required, and the tap strap would also be sampled again`. I assume this means that the test should verify that the tap straps are sampled again. Could this be added to the test? You could run the test twice: the first time you connect to the RISC-V (before the SW reset) and the second time to the LC_TAP but switch the straps to RISC-V before the SW reset and verify that the SW reset causes the straps to be re-sampled.

Reply:
The existing DV test doesn't test this, it just straps RV_DM tap and never changes it.

Reply:
I know that some DV tests are not complete unfortunately but I think I would favor writing a more complete SiVal test if possible instead of just doing what the DV tests does (if that does not require a device code change). Maybe @GregAC can let us know what he thinks since he is the testplan owner.

Reply:
I asked Greg for clarification and he thinks replicating what DV test does is good enough. We have a rv_dm_jtag_tap_sel test which tests that tap will be resampled on reset.